inc/utils/namespace.php

```diff
@@ -155,8 +194,49 @@ function query( array $query, array $params = [], string $path = '_search', stri
 	// Sanitize path.
 	$path = trim( $path, '/' );
 
+	$index_paths = [ 'analytics-*' ];
+
+	// Try to extract specific index names to query if possible.
+	if ( isset( $query['query']['bool']['filter'] ) && is_array( $query['query']['bool']['filter'] ) ) {
+		foreach ( $query['query']['bool']['filter'] as $filter ) {
+			if ( ! isset( $filter['range']['event_timestamp'] ) ) {
+				continue;
+			}
+
+			// Reset index paths.
+			$index_paths = [];
+
+			// Prevent fatal errors if an index doesn't exist.
+			$params['ignore_unavailable'] = 'true';
+
+			$from = absint( $filter['range']['event_timestamp']['gte'] ?? $filter['range']['event_timestamp']['gt'] ?? 0 );
+			$to = absint( $filter['range']['event_timestamp']['lte'] ?? $filter['range']['event_timestamp']['lt'] ?? milliseconds() );
+
+			$available_indices = get_indices();
+
+			foreach ( $available_indices as $index ) {
+				$day = strtotime( str_replace( 'analytics-', '', $index ) ) * 1000;
+				if ( $day >= $from && $day <= $to ) {
+					$index_paths[] = $index;
+				}
+			}
+
+			// Revert to all index if there are no matches.
+			if ( empty( $index_paths ) ) {
+				$index_paths[] = 'analytics-*';
+			}
+
+			break;
+		}
+	}
+
 	// Get URL.
-	$url = add_query_arg( $params, get_elasticsearch_url() . '/analytics*/' . $path );
+	$url = add_query_arg( $params, sprintf(
+		'%s/%s/%s',
+		get_elasticsearch_url(),
+		implode( ',', $index_paths ),
```

Comment:
Should we add any guards against too many selected indices here ?

Reply:
Yeah maybe. Elasticsearch allows up to 4kb in the URL - with 90 analytics indexes it's 1800 ascii characters (20 characters per index name) plus 89 commas.

Reply:
I'm not sure it's worth it looking at it, we cap the number of indexes at 90 and you'd get an error reported and a `null` value returned.

Reply:
If it exceeds the cap we can query for the wildcard instead, otherwise we'd still have the error/null return.

Reply:
True, I've updated it for that. I suppose if someone really really wanted to they could unschedule the index cleanup routine.